### DISCUSSION TITLE
How to sychronize editing of Corefiles with coredns reload plugin

### DISCUSSION THREAD

**Discussion by johlang-pillar** [coredns/coredns#5068] [Q&A] [answered]
Is there a way to synchronize changing of coredns config files with the reload plugin, or to trigger coredns to reload the config files without using the reload plugin?  I have a coredns setup that has external processes that modify the setup, and then use the reload plugin to allow coredns to detect the configuration has changed, and reload it.

The problem is if coredns is in the process of rereading the config at the same time the config file is being edited.  The main issue happens if removing lines from the config.  I think it is possible that when coredns rereads the Corefile and lines are deleted that it misses the reload plugin in at the end, and turns off reload checks.  For example in the simple case of:

import zone1.conf
import zone2.conf
import zone3.conf
. {
    log
    reload
}

If the line "import zone2.conf" is deleted at the same time coredns is reading the Corefile, is it possible that it will read the 'log' line and then read the next line which is not '}' and lose the 'reload' line, even though the new file will have it.

So I'm wondering if there is a different way to trigger a reload using a single to start it, instead of relying on the reload plugin.  Or would using the 'auto' plugin be more reliable?

**Comment by chrisohaver** [comment, marked as answer]
> Is there a way [...] to trigger coredns to reload the config files without using the reload plugin?

Yes.  You can send the CoreDNS process a SIGHUP or SIGUSR1.
